juniper_official/Subscriber-Service/address-pool-utilization.rule

```diff
@@ -0,0 +1,292 @@
+                        equal-to "$address-usage-ml" 1 {
+                            time-range 5m;
+                        }
+                    }
+                    then {
+                        status {
+                            color yellow;
+                            message "Pool $pool-name  ( Pool head : $linked-pool-head) utilization % ($address-usage-percent) is abnormal based on ML algorithm";
+                        }
+                    }
+                }
+                term pool-util-normal {
+                    then {
+                        status {
+                            color green;
+                            message "Pool $pool-name ( Pool head : $linked-pool-head) utilization % ($address-usage-percent) is normal";
+                        }
+                    }
+                }
+            }
+            variable filter-linked-pool-head {
+                value .*;
+                description "address pool header name";
+                type string;
+            }
+            variable filter-lsri-name {
+                value .*;
+                description "logical system routing instance name";
+                type string;
+            }
+            variable filter-pool-name {
+                value .*;
+                description "address pool name";
+                type string;
+            }
+            variable pool-utilization-max-threshold {
+                value 85;
+                description "address pool utilization maximum threshold value";
+                type int;
+            }
+            variable pool-utilization-min-threshold {
+                value 10;
+                description "address pool utilization minimum threshold value";
+                type int;
+            }
+        }
+        rule linked-address-pool {
+            synopsis "linked address pool utilization analyzer";
+            description "Monitors linked address pool usage percentage and notifies anomalies";
+            rule-frequency 90s;
+            function mean-pool-utilization {
+                path mean-pool-utilization.py;
+                method mean_pool_utilization;
+                argument list_pool_vector {
+                    mandatory;
+                }
+            }
+            field linked-pool-head {
+                constant {
+                    value "{{filter-linked-pool-head}}";
+                }
+                type string;
+                description "linked-pool-head name field";
+            }
+            field mean-pool-utilization {
+                formula {
+                    user-defined-function {
+                        function-name mean-pool-utilization;
+                        argument list_pool_vector "@list-pool-utilization";
+                    }
+                }
+                type integer;
+                description "average utilizaiton of linked address pool";
+            }
+            field mean-pool-utilization-ml {
```

Comment:
Can we rename this field to mean-pool-utilization-dynamic-threshold?

Reply:
Not able to put "dynamic-threshold" because of field length limit.